be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -203,7 +207,7 @@ bool ScanOperator::is_finished() const {
     }
 
     // Any io task is running or needs to run.
-    if (_num_running_io_tasks > 0 || !_morsel_queue->empty()) {
+    if (_num_running_io_tasks > 0 || _morsel_queue->has_more() || !_morsel_queue->empty()) {
```

Comment:
Is it enough for `has_output()` and `is_finished()` to adapt to the `has_more()` ?

Reply:
yes. only `is_finished()` is enough.